zen_document_parser/base.py

```diff
@@ -0,0 +1,231 @@
+from enum import IntEnum
+from pdfquery import PDFQuery
+from pdfminer.pdfdocument import PDFSyntaxError
+
+from zen_document_parser.exceptions import InvalidPDFError, UnknownVariantError
+
+
+class DocFieldType(IntEnum):
+    TEXT = 1
+    NUMBER = 2
+    CUSTOM = 3  # TODO: Forget this and have 'type' take a callable instead?
+
+
+class DocField(object):
+    def __init__(self, bbox, type=DocFieldType.TEXT, required=False,
+                 description=None):
+        self.bbox = bbox
+        self.type = type
```

Comment:
Yeah, I get that.
I thought about having it as `field_type`, but then the thought naming convention would have to lose consistency (field_type, required, description), or be redundant in all the params (field_description, field_required, etc.).

I guess it's okay to keep `type` an exception wrt naming conventions and rename it to `field_type`. Let me know which approach you wanna follow.